x-pack/plugins/elastic_assistant/server/lib/langchain/elasticsearch_store/elasticsearch_store.ts

```diff
@@ -44,17 +60,25 @@ export class ElasticsearchStore extends VectorStore {
   private readonly index: string;
   private readonly logger: Logger;
   private readonly model: string;
+  private readonly kbResource: string;
 
   _vectorstoreType(): string {
     return 'elasticsearch';
   }
 
-  constructor(esClient: ElasticsearchClient, index: string, logger: Logger, model?: string) {
+  constructor(
+    esClient: ElasticsearchClient,
+    index: string,
+    logger: Logger,
+    model?: string,
+    kbResource?: string | undefined
```

Comment:
Discussed scoping `kbResource` to esStore instances during pair review. Abstraction here makes sense and will instantiate additional scoped stores as needed, but could change in the future.